plugins/repository.go

```diff
@@ -0,0 +1,101 @@
+package plugins
+
+import (
+	"errors"
+	"github.com/spf13/cobra"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+const argRepository = "path"
+
+type RepositoryPlugin struct {
+	Plugin
+	Path string
+}
+
+func (p *RepositoryPlugin) IsEnabled() bool {
+	return p.Enabled
+}
+
+func (p *RepositoryPlugin) DefineCommandLineArgs(cmd *cobra.Command) error {
+	flags := cmd.Flags()
+	flags.String(argRepository, "", "scan repository folder")
+	return nil
+}
```

Comment:
I think that this only makes sense if we advance with your proposal approach, otherwise you only want to run confluence, for instance, and it gives the error message saying that the "required flag(s) "path" not set"

Reply:
Yes yes, but see this: https://github.com/Checkmarx/2ms/pull/53#discussion_r1172795965

Reply:
exactly